src/workerd/io/io-own.h

```diff
@@ -145,6 +157,30 @@ class DeleteQueue: public kj::AtomicRefcounted {
 private:
   template <typename T>
   SpecificOwnedObject<T>* addObjectImpl(kj::Own<T> obj, OwnedObjectList& ownedObjects);
+
+  kj::Promise<void> resetCrossThreadSignal();
+
+  friend class IoContext;
+};
+
+// Object which can push actions into a specific DeleteQueue then signal it's
+// owning IoContext to wake up to process the queue. This is a bit of a hack of
+// the DeleteQueue concept that allows us to use the same queue for more than
+// just deletions.
+class IoDeferral {
+public:
+  IoDeferral(kj::Own<const DeleteQueue> deleteQueue): deleteQueue(kj::mv(deleteQueue)) {}
+
+  // Defers the specified action to the owning IoContext. The next time the
```

Comment:
This comment implies that we won't proactively wake up the target context, but instead just wait until it happens to be entered for some other reason. But actually, we do proactively wake it up.

I think maybe this class should be called `IoCrossContextExecutor` and the `defer()` method should be called `execute()`.